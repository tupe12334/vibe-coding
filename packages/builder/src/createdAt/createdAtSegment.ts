import type { RootContent } from "mdast";

/**
 * Generate a "Created at" timestamp segment.
 *
 * @returns markdown AST nodes representing the timestamp
 */
export const createdAtSegment = async (date: Date = new Date()): Promise<RootContent[]> => {
  const now = date.toISOString();
  const segment: RootContent[] = [
    {
      type: "paragraph",
      children: [
        { type: "text", value: `Created at: ${now}` },
      ],
    },
  ];

  return segment;
};
