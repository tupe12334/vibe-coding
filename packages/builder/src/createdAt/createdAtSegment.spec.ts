import { describe, expect, test } from "vitest";
import { createdAtSegment } from "./createdAtSegment";

describe("createdAtSegment", () => {
  test("returns timestamp segment", async () => {
    const date = new Date("2020-01-01T00:00:00.000Z");
    const segment = await createdAtSegment(date);
    expect(segment).toMatchSnapshot();
  });
});
