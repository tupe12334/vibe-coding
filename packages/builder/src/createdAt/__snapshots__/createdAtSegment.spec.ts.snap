// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`createdAtSegment > returns timestamp segment 1`] = `
[
  {
    "children": [
      {
        "type": "text",
        "value": "Created at: 2020-01-01T00:00:00.000Z",
      },
    ],
    "type": "paragraph",
  },
]
`;
